### ISSUE TITLE
Action fails with conflicting versions of sysdeps

### ISSUE THREAD

**Issue by mpadge** [ropensci-review-tools/pkgcheck-action#48]
@assignUser @jooolia This issue is to explore whether there might be solutions to https://github.com/lifewatch/mregions2/issues/24, although my thought is that the problem there can not actually be resolved. It arises within {pak}'s attempts to install the lockfile, because librdf now requires libraptor2-dev which requires libcurl4 via `libcurl4-gnutls-dev` and not `libcurl4-openssl-dev`, in spite of [clearly stating that it works with both](https://packages.ubuntu.com/noble/libraptor2-dev). Because `pak` relies on a *lock* file, it refuses to change this system installation and so errors. I tried switching out the `openssl` version for `gnutls` in the container itself, but there are then other system dependencies which try to switch that back to `openssl`, and so still error.

@assignUser Do you have any ideas here? We basically need a way to tell `pak:install_lockfile()` to ignore specific [system dependencies](https://github.com/r-lib/pak/blob/cc9ddb76c6b9d4911543d34903e7fa96d29f13f6/R/lockfile.R#L100) and proceed with the rest regardless. As far as I understand it, I think there is no way to resolve the issue.

Everything that pulls in `curl` at R level or `libcurl` at system level ultimately reads [this hard-coded need for `libcurl4-openssl-dev`](https://github.com/rstudio/r-system-requirements/blob/3a955f87881517c9332677a359af978de68466cf/rules/libcurl.json#L5), even though alternative implementations like `libcurl4-gnutls-dev` work (mostly) fine. Every time that's run, it will uninstall any of these alternative implementations. Then when `librdf0-dev` is installed, it will try through `libraptor2-dev` to uninstall the `openssl` version and replace with `gnutls` version. That all works fine in any system with ability to overwrite system libraries (albeit with potentially unpredictable behaviour of libcurl, but that's another issue ...), but is impossible to resolve within any kind of lockfile system. And at that point, I'm stuck. We can't have a lockfile which allows `librdf0-dev` and simultaneously all other packages and system requirements which depend on `libcurl4-openssl-dev`.

---

Hmmm... even that's not quite right. Running `apt-get install libraptor2-dev` (in a Ubuntu 22.04 container) states that `libcurl4-openssl-dev` will be REMOVED, but actually what it does is leave `/usr/lib/x86_64-linux-gnu/libcurl.so.4` in place and create an alternative (via [virtual package](https://packages.ubuntu.com/noble/libcurl-ssl-dev)) installation of `/usr/lib/x86_64-linux-gnu/libcurl-gnutls.so`. So both installs are kept, and should be able to be used via the virtual package, and yet {pak}'s attempts to run `apt-get install librdf0-dev` via the lockfile still fail.

**Comment by assignUser** [comment]
Hm, tricky. 

> As far as I understand it, I think there is no way to resolve the issue.

Yeah I think this might require a fix in {pak}/r-system-requirement though it's unclear to me what a solution would look like if the  apt dependency resolution seems broken for libraptor2-dev? Sorry nothing comes to mind immediately :(

**Comment by mpadge** [comment]
Thanks @assignUser. I would have liked to find a "proper" solution for this through some kind of dependency management resolution, but that seems not possible here, alas. I'll implement the unsatisfying work-around of pre-installing the `rdflib` R package, which does the double-install outside of {pak}, and properly links all virtual package paths to the duplicated system libs.

**Comment by assignUser** [comment]
> unsatisfying work-around

Yeah, I was thinking that that might work but yeah unsatisfying is an apt (:rofl:) description.
